src/server/middleware/nydus/nydus.go

```diff
@@ -0,0 +1,142 @@
+    "config": {
+      "mediaType": "application/vnd.oci.image.config.v1+json",
+      "digest": "sha256:f7d0778a3c468a5203e95a9efd4d67ecef0d2a04866bb3320f0d5d637812aaee",
+      "size": 466
+    },
+    "layers": [
+      {
+        "mediaType": "application/vnd.oci.image.layer.nydus.blob.v1",
+        "digest": "sha256:fd9923a8e2bdc53747dbba3311be876a1deff4658785830e6030c5a8287acf74 ",
+        "size": 3011,
+        "annotations": {
+          "containerd.io/snapshot/nydus-blob": "true"
+        }
+      },
+      {
+        "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+        "digest": "sha256:d49bf6d7db9dac935b99d4c2c846b0d280f550aae62012f888d5a6e3ca59a589",
+        "size": 459,
+        "annotations": {
+            "containerd.io/snapshot/nydus-blob-ids": "[\"fd9923a8e2bdc53747dbba3311be876a1deff4658785830e6030c5a8287acf74\"]",
+            "containerd.io/snapshot/nydus-bootstrap": "true",
+            "containerd.io/snapshot/nydus-rafs-version": "5"
+        }
+      }
+    ],
+    "annotations": {
+        "io.goharbor.artifact.v1alpha1.acceleration.driver.name":"nydus",
+        "io.goharbor.artifact.v1alpha1.acceleration.driver.version":"5",
+        "io.goharbor.artifact.v1alpha1.acceleration.source.digest":"sha256:f54a58bc1aac5ea1a25d796ae155dc228b3f0e11d046ae276b39c4bf2f13d8c4"
+    }
+}
+
+*/
+func AcceleratorMiddleware() func(http.Handler) http.Handler {
+	return middleware.AfterResponse(func(w http.ResponseWriter, r *http.Request, statusCode int) error {
+		if statusCode != http.StatusCreated {
+			return nil
+		}
+
+		log.Debug("Start NydusAccelerator Middleware")
+		ctx := r.Context()
+		logger := log.G(ctx).WithFields(log.Fields{"middleware": "nydus"})
+
+		none := lib.ArtifactInfo{}
+		info := lib.GetArtifactInfo(ctx)
+		if info == none {
+			return errors.New("artifactinfo middleware required before this middleware").WithCode(errors.NotFoundCode)
+		}
+		if info.Tag == "" {
+			return nil
+		}
+
+		body, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return err
+		}
+
+		contentType := r.Header.Get("Content-Type")
+		manifest, desc, err := distribution.UnmarshalManifest(contentType, body)
+		if err != nil {
+			logger.Errorf("unmarshal manifest failed, error: %v", err)
+			return err
+		}
+
+		var isNydus bool
+		for _, descriptor := range manifest.References() {
+			annotationMap := descriptor.Annotations
+			if _, ok := annotationMap["containerd.io/snapshot/nydus-bootstrap"]; ok {
+				if annotationMap["containerd.io/snapshot/nydus-bootstrap"] == "true" {
+					isNydus = true
+					break
+				}
+			}
+		}
+		log.Debug("isNydus: ", isNydus)
+
+		_, payload, err := manifest.Payload()
+		if err != nil {
+			return err
+		}
+		mf := &v1.Manifest{}
+		if err := json.Unmarshal(payload, mf); err != nil {
+			return err
+		}
+
+		if isNydus {
+			subjectArt, err := artifact.Ctl.GetByReference(ctx, info.Repository, mf.Annotations["io.goharbor.artifact.v1alpha1.acceleration.source.digest"], nil)
```

Comment:
The result from `distribution.UnmarshalManifest` can not get this `io.goharbor.artifact.v1alpha1.acceleration.source.digest` annotation?

Reply:
`io.goharbor.artifact.v1alpha1.acceleration.source.digest` mark the subject/source artifact digest and it returns from nydus

Reply:
`distribution.UnmarshalManifest` manifest reference contains the descriptor of each layers, but the annotation we need to got is from the payload., Thus we got the `manifest.Payload()`, then Unmarshal it.